analysis_options.yaml

```diff
@@ -13,12 +13,12 @@ analyzer:
     # Please see https://github.com/flutter/flutter/pull/24528 for details.
```

Comment:
Would like to propose adding three new options

```
    public_member_api_docs: true
    use_super_parameters: true
    require_trailing_commas: true
```

**Why?**

1. `public_member_api_docs` 

Will help encourage contributions to document any changes they make in PRs, found this to be typically really useful in projects I've worked on where I'm looking back at code I wrote 6+ months ago and it's far easier to understand with good, meaningful comments

I wouldn't worry so much to document **everything** in this PR but anything new should be commented and we can gradually document more later

2. `use_super_parameters` 

Always think it's good to adopt the latest features introduced in newer versions of Dart, been using this super parameters in other projects and it helps (imo) make the code a lot cleaner 

3. `require_trailing_commas` 

Another one to help improve readability - it helps avoid code from going on to another line and not being nicely formatted

Reply:
adopting the `use_super_parameters` lint rule may require a dart SDK version bump